bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/handler/IPBridgeThingHandler.java

```diff
@@ -56,7 +59,45 @@ public IPBridgeThingHandler(Bridge bridge, NetworkAddressService networkAddressS
 
     @Override
     public void initialize() {
+        // initialisation would take too long and show a warning during binding startup
+        // KNX secure is adding serious delay
+        updateStatus(ThingStatus.UNKNOWN);
+        initJob = scheduler.submit(() -> {
+            initializeLater();
+        });
+    }
+
+    public void initializeLater() {
         IPBridgeConfiguration config = getConfigAs(IPBridgeConfiguration.class);
+        boolean securityAvailable = false;
+        try {
+            securityAvailable = initializeSecurity(config.getRouterBackboneKey(),
+                    config.getTunnelDeviceAuthentication(), config.getTunnelUserId(), config.getTunnelUserPassword());
+            if (securityAvailable) {
+                logger.debug("KNX secure: router backboneGroupKey is {} set",
+                        ((secureRouting.backboneGroupKey.length == 16) ? "properly" : "not"));
+                boolean tunnelOk = ((secureTunnel.user > 0) && (secureTunnel.devKey.length == 16)
+                        && (secureTunnel.userKey.length == 16));
+                logger.debug("KNX secure: tunnel keys are {} set", (tunnelOk ? "properly" : "not"));
+            } else {
+                logger.debug("KNX security not configured");
+            }
+        } catch (NullPointerException e) {
```

Comment:
You should not catch `NullPointerException`. If this is a bug within the binding, it should be fixed. If it's a bug within an external library outside our control, you can surround API call by try/catch and catch `Exception`.

Reply:
I know this looks a bit strange, but actually this is some kind of graceful degrading.
The extended version of this PR for data secure contains reading a KNX keyring file within `initializeSecurity()` (using Calimero library). I think I have seen NPEs out of my control in the past, this is why I added this catch block. 
The effect is basically that the addon is running without security support instead of crashing.

But I think it is also fine to remove this. WDYT?

Reply:
If `initializeSecurity()` makes an API call to Calimero, I would propose to explicitly guard that call:
```
try {
    callCalimero();
}
catch (Exception e) {
    // This is a bug, handle gracefully - just as example:
    throw KnxException("Unexpected Calimero exception", e)
}
```

Here and below I would then explicitly catch checked exceptions from own code and documented runtime exceptions from Calimero if they are not mapped/rethrown from own code.

In other words: Yes, it would be fine to remove it, if at the same time making sure we have guards elsewhere to make sure the binding works reliably even if Calimero misbehaves.

Reply:
Ok, will use this pattern for the extended PR to guard the call to the base library. This is anyway not part of this PR.

`initializeSecurity()` now only throws `KNXSecureException`. I checked the function calls within `initializeSecurity` again and currently everything seems to be covered by catching `KNXSecureException`.

Reply:
Small correction to example - you probably get the point, but now reading it again, it must be corrected. :)

```
try {
    callCalimero();
}
catch (KnxException e) { // and others expected
    throw e; // if not handled here
}
catch (Exception e) {
    // This is a bug, handle gracefully - just as example:
    logger.trace("", e);
    throw KnxException("Unexpected Calimero exception", e)
}
```